arch/wec7/ua_clock.c

```diff
@@ -0,0 +1,63 @@
+/* This work is licensed under a Creative Commons CCZero 1.0 Universal License.
+ * See http://creativecommons.org/publicdomain/zero/1.0/ for more information.
+ *
+ *    Copyright 2018 (c) Stephan
+ */
+
+#ifdef UA_ARCHITECTURE_WEC7
+
+#ifndef _BSD_SOURCE
+# define _BSD_SOURCE
+#endif
+
+#include "ua_types.h"
+#include <time.h>
+/* Backup definition of SLIST_ENTRY on mingw winnt.h */
+# ifdef SLIST_ENTRY
+#  pragma push_macro("SLIST_ENTRY")
+#  undef SLIST_ENTRY
+#  define POP_SLIST_ENTRY
+# endif
+# include <windows.h>
+/* restore definition */
+# ifdef POP_SLIST_ENTRY
+#  undef SLIST_ENTRY
+#  undef POP_SLIST_ENTRY
+#  pragma pop_macro("SLIST_ENTRY")
+# endif
+
+UA_DateTime UA_DateTime_now(void) {
+    /* Windows filetime has the same definition as UA_DateTime */
+    FILETIME ft;
+    SYSTEMTIME st;
+    GetSystemTime(&st);
+    SystemTimeToFileTime(&st, &ft);
+    ULARGE_INTEGER ul;
+    ul.LowPart = ft.dwLowDateTime;
+    ul.HighPart = ft.dwHighDateTime;
+    return (UA_DateTime)ul.QuadPart;
+}
+#ifndef UNDER_CE
+// The problem here is that not all time functions are implemented in Windows CE.
+UA_Int64 UA_DateTime_localTimeUtcOffset(void) {
```

Comment:
As a workaround you could just return 0 for the offset and mention it in a comment that the local time offset is always 0 here.